src/main/java/org/folio/services/domainevent/ItemDomainEventService.java

```diff
@@ -0,0 +1,163 @@
+package org.folio.services.domainevent;
+
+import static io.vertx.core.Future.succeededFuture;
+import static org.apache.logging.log4j.LogManager.getLogger;
+import static org.folio.rest.support.ResponseUtil.isCreateSuccessResponse;
+import static org.folio.rest.support.ResponseUtil.isDeleteSuccessResponse;
+import static org.folio.rest.support.ResponseUtil.isUpdateSuccessResponse;
+import static org.folio.services.kafka.topic.KafkaTopic.INVENTORY_ITEM;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import javax.ws.rs.core.Response;
+
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.ImmutableTriple;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.commons.lang3.tuple.Triple;
+import org.apache.logging.log4j.Logger;
+import org.folio.persist.HoldingsRepository;
+import org.folio.persist.ItemRepository;
+import org.folio.rest.jaxrs.model.HoldingsRecord;
+import org.folio.rest.jaxrs.model.Item;
+import org.folio.services.batch.BatchOperationContext;
+import org.folio.services.item.effectivevalues.ItemWithHolding;
+
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+
+public class ItemDomainEventService {
+  private static final Logger log = getLogger(ItemDomainEventService.class);
+
+  private final ItemRepository itemRepository;
+  private final HoldingsRepository holdingsRepository;
+  private final BaseDomainEventService<Item> domainEventService;
+
+  public ItemDomainEventService(Context context, Map<String, String> okapiHeaders) {
+    itemRepository = new ItemRepository(context, okapiHeaders);
+    holdingsRepository = new HoldingsRepository(context, okapiHeaders);
+    domainEventService = new BaseDomainEventService<>(context, okapiHeaders, INVENTORY_ITEM);
+  }
+
+  public Function<Response, Future<Response>> itemUpdated(String instanceId, Item oldItem) {
```

Comment:
What does this method do? Does it publish events when an item has been updated?

Reply:
Yes, this is the main its responsibility.

Reply:
Maybe the name of the method or class could reveal that intent more.

```suggestion
  public Function<Response, Future<Response>> publishDomainEventsOnItemUpdated(String instanceId, Item oldItem) {
```

```suggestion
  public Function<Response, Future<Response>> publishItemUpdatedDomainEvent(String instanceId, Item oldItem) {
```

Or the class could be renamed to `ItemDomainEventPublisher` and the method to `onItemUpdated` or `publishItemUpdated`

```suggestion
  public Function<Response, Future<Response>> onItemUpdated(String instanceId, Item oldItem) {
```

```suggestion
  public Function<Response, Future<Response>> publishItemUpdated(String instanceId, Item oldItem) {
```

Reply:
I've renamed the classes to *Publisher and added `publish` prefix to the methods.